pyroclast/common/preprocessed_dataset.py

```diff
@@ -0,0 +1,30 @@
+import tensorflow as tf
+
+
+class PreprocessedDataset(object):
+
+    def __init__(self, ds, module, filepath, key='image'):
+        self.module = module
+        self.base_ds = ds
+        self.path = filepath
+
+        # TODO: only save if it doesn't already exist
```

Comment:
While hashing can be pushed off to later, we should probably have this TODO done before merging.